### PR TITLE
vendor: dockerfile parser from buildkit

### PR DESCRIPTION
**Pull request by tonistiigi**
Dockerfile spec helper packages were cut from https://github.com/moby/moby/tree/71cd53e4a197b303c6ba086bd584ffd67a884281/builder/dockerfile in  https://github.com/moby/buildkit/pull/425 and are now vendored back so we don't have two diverging sources of truth.

### CODE REVIEW

**Review by AkihiroSuda — APPROVED** [review]
LGTM if green

**Review by thaJeztah — APPROVED** [review]
LGTM, but let's give some more maintainers a quick look before we merge

**Review by vdemeester — APPROVED** [review]
LGTM :lion: :tada:

**Pull request merged**
